ArtOfIllusion/src/artofillusion/ArtOfIllusion.java

```diff
@@ -371,51 +367,50 @@ public static void quit()
       if (windows.contains(win))
         return;
     }
-    List plugins = PluginRegistry.getPlugins(Plugin.class);
-    for (int i = 0; i < plugins.size(); i++)
+    
+    for (Plugin plugin: PluginRegistry.getPlugins(Plugin.class))
     {
       try
       {
-        ((Plugin) plugins.get(i)).processMessage(Plugin.APPLICATION_STOPPING, new Object [0]);
+        plugin.processMessage(Plugin.APPLICATION_STOPPING, new Object [0]);
       }
       catch (Throwable tx)
       {
         tx.printStackTrace();
-        String name = plugins.get(i).getClass().getName();
-        name = name.substring(name.lastIndexOf('.')+1);
-        new BStandardDialog("", UIUtilities.breakString(Translate.text("pluginNotifyError", name)), BStandardDialog.ERROR).showMessageDialog(null);
+        new BStandardDialog("", UIUtilities.breakString(Translate.text("pluginNotifyError", plugin.getClass().getSimpleName())), BStandardDialog.ERROR).showMessageDialog(null);
       }
     }
     System.exit(0);
   }
 
   /** Execute all startup scripts. */
-
+  @SuppressWarnings("CallToPrintStackTrace")
   private static void runStartupScripts()
   {
     String files[] = new File(STARTUP_SCRIPT_DIRECTORY).list();
+    if(null == files) return;
```

Comment:
I know it's just nitpicking, but can we maintain a consistent code style?  I think that helps readability.  That should be

```Java
if (null == files)
  return;
```